index.bs

```diff
@@ -261,39 +259,93 @@ vrDisplay.requestPresent([{ source: canvas }]).then(function() {
+  void setLeftBounds(float left, float bottom, float right, float top);
+  sequence&lt;float&gt; getLeftBounds();
+
+  void setRightBounds(float left, float bottom, float right, float top);
+  sequence&lt;float&gt; getRightBounds();
 };
 
-dictionary VRLayerInit {
-  VRSource? source = null;
+interface CanvasVRLayerReadOnly : VRLayerReadOnly {
+  readonly attribute CanvasVRSource source;
+
+  readonly attribute float renderScale;
+  readonly attribute boolean resizeSourceOnPresent;
 
-  sequence&lt;float&gt; leftBounds = [ ];
-  sequence&lt;float&gt; rightBounds = [ ];
+  readonly attribute unsigned long renderWidth;
+  readonly attribute unsigned long renderHeight;
+
+  sequence&lt;float&gt; getLeftBounds();
+  sequence&lt;float&gt; getRightBounds();
 };
 </pre>
 
-### Attributes ### {#vrlayer-attributes}
+### Attributes ### {#canvasvrlayer-attributes}
+
+<dfn attribute for="CanvasVRLayer">source</dfn>
+The {{CanvasVRLayer/source}} attribute defines the canvas whose contents will be presented for this layer by the {{VRDisplay}} when {{VRDisplay}}.{{submitFrame()}} is called. Upon being passed into {{requestPresent()}} the current backbuffer of the source's context MAY be lost, even if the context was created with the <code>preserveDrawingBuffer</code> context creation attribute set to <code>true</code>. Calls to {{requestPresent()}} MUST be rejected if the {{CanvasVRLayer/source}} of any {{CanvasVRLayer}} provided does not represent a canvas with a <code>webgl</code> or <code>webgl2</code> context.
+
+<dfn attribute for="CanvasVRLayer">renderScale</dfn>
+Specifies the ratio of the number of canvas pixels to the {{VRDisplay}}'s physical pixels at the center of the output image. Note that a value of <code>1.0</code> will generally result in a larger canvas resolution than the {{VRDisplay}}'s physical screen in order to allow for corrective distiorion. Specifying a lower value can improve performance. A value of <code>0.0</code> instructs the user agent to select a ratio that provides a good balance between quality and performance for most applications on the current {{VRDisplay}}. Calls to {{requestPresent()}} MUST be rejected if the renderScale of any {{CanvasVRLayer}} provided is less than <code>0.0</code>.
+
+If this value was set to <code>0.0</code> when the {{CanvasVRLayer}} is passed to {{requestPresent()}} the {{renderScale}} of the corresponding {{CanvasVRLayerReadOnly}} returned from {{getLayers()}} should be set to the value selected by the user agent.
+
+The default value for {{renderScale}} MUST be <code>0.0</code>.
+
+<dfn attribute for="CanvasVRLayer">resizeSourceOnPresent</dfn>
+If <code>true</code> the {{CanvasVRLayer/source}} of any {{CanvasVRLayer}} provided to {{requestPresent()}} will be resized to the width and height computed from the {{renderScale}} before the promise resolves or the vrdisplaypresentchange events are fired. If {{requestPresent()}} is rejected the {{CanvasVRLayer/source}} dimensions will not be changed.
 
```

Comment:
WindowsHolographic has a similar property to renderScale called [ViewportScaleFactor](https://msdn.microsoft.com/en-us/library/windows/apps/windows.graphics.holographic.holographiccamera.viewportscalefactor.aspx). Here is what the docs have to say:   
`Each frame, the system decides the size of the viewport within the render target that the app must render that frame. Sometimes the system may lower this to free up system resources, for example when video capture is occurring. Regardless of the viewport size, the rendered image is always scaled up to the full size of the displays. An app looking to lower its rendering cost for other reasons can request a smaller viewport.`

`You can specify a scale factor between 0.0 and 1.0, which is multiplied by the default viewport size. In future frames, the system will specify that you render to the nearest valid viewport size to this requested size.`

This means that the viewport you're given depends on:

1. What the system decides is best for you based upon some the capabilities of the system and some other operation happening in the system that the developer can't control or have knowledge about.
2. A rounding of the requested viewport scale with valid viewport sizes. 

I still feel that the browser should not be resizing the framebuffer based on what is happening on the webpage.

Given this, I would prefer that we make the following changes:
1. Change `renderScale` to `requestedRenderScale`
2. Add the notion of a viewport size that the web developer needs to render to within the Canvas.  

What is the rationale for setting resizeSource to false?  Do we expect that web developers will want to super size their content?  If so, why can't they do this themselves via an intermediate framebuffer?  Seems like having resizeSource=true as the only option would be a simplification we can make to the API.  Same questions apply to renderScale > 1.0.

What do you think?